colorizer_data/writer.py

```diff
@@ -381,18 +394,50 @@ def write_manifest(
 
         Must be called **AFTER** all other data is written.
 
+        Args:
+            num_frames (int): DEPRECATED. Define to generate the expected paths for frame images.
+            metadata (ColorizerMetadata): Metadata to be written with the dataset. Leave fields blank to use existing default values.
+
+        Note that some metadata fields (like `last_modified`, `_writer_version`, `_revision`, and `date_created`) will
+        be automatically updated. Add definitions for these fields in the `metadata` argument to override this behavior.
+
         [documentation](https://github.com/allen-cell-animated/colorizer-data/blob/main/documentation/DATA_FORMAT.md#Dataset)
         """
 
         if num_frames is not None and self.manifest["frames"] is None:
-            logging.warn(
+            logging.warning(
                 "ColorizerDatasetWriter: The argument `num_frames` on `write_manifest` is deprecated and will be removed in the future! Please call `set_frame_paths(generate_frame_paths(num_frames))` instead."
             )
             self.set_frame_paths(generate_frame_paths(num_frames))
 
-        # Add the metadata
-        if metadata:
-            self.manifest["metadata"] = metadata.to_json()
+        # Automatically update metadata fields. This behavior can be overwritten using
+        # by defining the relevant fields in the the `metadata` argument.
+        current_time = datetime.now(timezone.utc).strftime(DATETIME_FORMAT)
+        # Update creation date if missing
+        if self.metadata.date_created == None:
+            self.metadata.date_created = current_time
+        # Update revision number
+        revision = self.metadata._revision
+        if revision == None:
+            self.metadata._revision = 0
+        else:
+            self.metadata._revision = revision + 1
+        # Update data version + modified timestamp
+        self.metadata.last_modified = current_time
+        self.metadata._writer_version = CURRENT_VERSION
+
+        # Use default dataset name from writer constructor if no name was loaded
+        # (will be overridden by `metadata.name` argument if provided)
+        if self.metadata.name == None:
+            self.metadata.name = self.default_dataset_name
+
+        # Optionally merge new metadata with old
+        if metadata != None:
```

Comment:
We may want to use `is None` and `is not None` instead of `==` and `!=`, unless you have a specific reason for using the latter.

Reply:
Fixed, thank you!